cloud/amazon/public/resources/instanceprofile.go

```diff
@@ -185,10 +186,23 @@ func (r *InstanceProfile) Apply(actual, expected cloud.Resource, immutable *clus
 		logger.Debug("CreateInstanceProfile error: %v", err)
 		if err.(awserr.Error).Code() != iam.ErrCodeEntityAlreadyExistsException {
 			return nil, nil, err
+		}else {
+			profileinput := &iam.GetInstanceProfileInput{
+				InstanceProfileName: &expected.(*InstanceProfile).Name,
+			}
+			outInstanceProfile, err := Sdk.IAM.GetInstanceProfile(profileinput)
```

Comment:
+1 to this - I was thinking of submitting a patch fixing this specifically.

Reply:
please do I will rebase

Reply:
I just rebased on master but if you want to PR this gopher it